package/base-files/files/lib/functions/leds.sh

```diff
@@ -11,13 +11,42 @@ get_dt_led_path() {
 	echo "$ledpath"
 }
 
+get_dt_led_color_func() {
+	local enum
+	local func
+	local idx
+	local label
+
+	[ -e "$1/function" ] && func=$(cat "$1/function")
+	[ -e "$1/color" ] && idx=$((0x$(hexdump -n 4 -e '4/1 "%02x"' "$1/color")))
+	[ -e "$1/function-enumerator" ] && \
+		enum=$((0x$(hexdump -n 4 -e '4/1 "%02x"' "$1/function-enumerator")))
+
+	[ -z "$idx" -o -z "$func" ] && return 2
```

Comment:
shellchecker:

```
[ -z "$idx" -o -z "$func" ] && return 2
            ^-- [SC2166](https://www.shellcheck.net/wiki/SC2166) (warning): Prefer [ p ] || [ q ] as [ p -o q ] is not well defined.
```

Reply:
Thanks for your review. I searched the Linux led code and found that one of `function` and `color` property could be missing. So I changed some logic to match this behavior. I haven't tested the new changes yet, but it shouldn't have introduced any other errors.

The ref:
https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/drivers/leds/led-core.c?h=next-20231201#n501

The git force-push diff:
https://github.com/openwrt/openwrt/compare/9d2262c4fc44fa253b7f65da12d5fee5168c1476..ac049e3ab5adc5ff025d466c382da931846745e3

Reply:
Test result shows all LEDs still work.
```
dts:
	aliases {
		led-boot = &led_status_red;
		led-failsafe = &led_status_blue;
		led-running = &led_status_yellow;
		led-upgrade = &led_status_blue;
	};

	leds {
		compatible = "gpio-leds";

		led_status_blue: led-blue {
			color = <LED_COLOR_ID_BLUE>;
			gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
		};

		led_status_red: led-red {
			function = LED_FUNCTION_STATUS;
			function-enumerator = <0>;
			gpios = <&gpio0 8 GPIO_ACTIVE_HIGH>;
		};

		led_status_yellow: led-yellow {
			color = <LED_COLOR_ID_YELLOW>;
			function = LED_FUNCTION_STATUS;
			function-enumerator = <100>;
			gpios = <&gpio0 11 GPIO_ACTIVE_LOW>;
		};
	};


root@OpenWrt:~# ls /sys/class/leds/
:status-0                mt76-phy0                rt2800soc-phy1::quality  yellow:status-100
blue:                    rt2800soc-phy1::assoc    rt2800soc-phy1::radio
root@OpenWrt:~# ls /sys/firmware/devicetree/base/leds/led-blue/
color  gpios  name
root@OpenWrt:~# ls /sys/firmware/devicetree/base/leds/led-red/
function             function-enumerator  gpios                name
root@OpenWrt:~# ls /sys/firmware/devicetree/base/leds/led-yellow/
color                function             function-enumerator  gpios                name
```